kite-data/kite-data-mapreduce/src/main/java/org/kitesdk/data/mapreduce/DatasetKeyInputFormat.java

```diff
@@ -227,6 +226,7 @@ public Configuration getConf() {
   @Override
   public void setConf(Configuration configuration) {
     conf = configuration;
+    DefaultConfiguration.set(conf);
```

Comment:
You know the code here far better than I, I'm assuming there's no worries of concurrency issues when you set the global static conf here? Like some other thread doing Kite related things that doesn't expect DatasetKeyInputFormat to have overridden the global?